### PR TITLE
Add shortcuts to toggle antialiasing

### LINKED ISSUE #26: Feature: Toggle antialiasing
The current situation (disabling antialiasing after a zoom threshold) is a good default, but sometimes I'd like to manually enable/disable antialiasing with a keyboard shortcut or a button.

### PR DESCRIPTION
**Pull request by Aloso**
This adds shortcuts to toggle antialiasing. The current value is stored in the cache file. There are three possible values:

* `always` – turn antialiasing on
* `never` – turn antialiasing off
* `auto` – antialiasing depends on the zoom factor

There are two new key bindings:

* `toggle_antialias` (default <kbd>S</kbd>): Sets antialiasing to `always` or `never`
* `automatic_antialias` (default <kbd>Alt+S</kbd>): Sets antialiasing to `auto`

The `cfg.toml` has a new setting `image.antialiasing`, which can have the following values:

* `always`, `never`, `auto` – see above
* `previous` – restores the setting from the last session

Closes #26

### CODE REVIEW

**Comment by Aloso** [comment]
> Currently this only applies to magnificaiton. I'm wondering if it would be a good idea to apply this to minification as well.

How do you mean? When pressing <kbd>S</kbd>, antialiasing is toggled irrespective of the zoom level. Or do you mean something different?

> I think that there could be a cfg field for remembering the aa setting.

Now that I think of it, I agree that a config setting would be better.

> I like the initiative to more compactly fit the contents of the usage panel; but for me the previous layout was definitely cleaner.

I'll change it back then.

**Comment by ArturKovacs** [comment]
What I meant by minification was that when the image is shrunk we still use smoothing even if aa is off. But I honestly don't know if anyone would want to turn that off so let's just leave it as it is. If someone needs that, they'll open an issue.

> Now that I think of it, I agree that using config.toml would be better than cache.toml.

To clarify, I was thinking of using both. The cache would remain the way you made it and the cfg would have an optional field called `use_last_aa` which is a bool which is `false` by default. It's also because emulsion shouldn't write to the cfg file.

**Comment by Aloso** [comment]
@ArturKovacs I think I addressed all of your concerns.

File: src/picture_widget.rs [hunk 1]
```diff
@@ -197,6 +200,21 @@ impl PictureWidgetData {
 		self.render_validity.invalidate();
 	}
 
+	pub fn toggle_antialias(&mut self) {
+		let aa = match self.antialiasing {
+			Antialias::Auto if self.img_texel_size < 4f32 => Antialias::Never,
```
Comment (ArturKovacs) [inline src/picture_widget.rs:205]: I just realized that the constant `4f32` appears twice in the code. We should introduce a const for it, like

```rust
const ANTIALIAS_TEXEL_THRESHOLD: f32 = 4f32;
```

**Comment by ArturKovacs** [comment]
Thanks!

**Pull request merged**